test/services/prisonerServiceTest.js

```diff
@@ -310,42 +338,53 @@ describe('prisonerDetailsService', () => {
 
     const sortingScenarios = [
       {
-        label: 'only one id in many relations',
-        relationships: [{ personId: '' }, { personId: null }, { personId: '1' }],
-        selected: '1',
+        label: 'only one person id, use that one',
+        relationships: [{ personId: '' }, { personId: null }, { personId: 1 }],
+        identifiers: [[{ identifierType: 'EXTERNAL_REL', identifierValue: 'delius1' }]],
+        selected: 'delius1',
       },
       {
-        label: 'highest number',
-        relationships: [{ personId: 222 }, { personId: 333 }, { personId: 111 }],
-        selected: 333,
+        label: 'only one delius id, use that one',
+        relationships: [{ personId: 1 }, { personId: 2 }, { personId: 3 }],
+        identifiers: [
+          [{ identifierType: 'EXTERNAL_REL', identifierValue: null }],
+          [{ identifierType: 'EXTERNAL_REL', identifierValue: 'delius2' }],
+          [{}],
+        ],
+        selected: 'delius2',
       },
       {
-        label: 'highest string number',
-        relationships: [{ personId: '299' }, { personId: '301' }, { personId: '1' }],
-        selected: '301',
+        label: 'multiple delius ids, use highest person id (NB not highest delius id)',
+        relationships: [{ personId: '1' }, { personId: '2' }, { personId: '3' }],
+        identifiers: [
+          [{ identifierType: 'EXTERNAL_REL', identifierValue: '999' }],
+          [{ identifierType: 'EXTERNAL_REL', identifierValue: '1' }],
+          [{ identifierType: 'EXTERNAL_REL', identifierValue: '-1' }],
+        ],
+        selected: '-1',
       },
       {
-        label: 'highest string',
-        relationships: [{ personId: 'caaa' }, { personId: 'bzzz' }, { personId: 'a999' }],
-        selected: 'caaa',
-      },
-      {
-        label: 'only one relation',
-        relationships: [{ personId: 'c' }],
-        selected: 'c',
-      },
-      {
-        label: 'either one if same',
-        relationships: [{ personId: 2 }, { personId: 2 }, { personId: 2 }, { personId: 2 }],
-        selected: 2,
+        label: 'multiple delius ids, use either one if same person id',
+        relationships: [{ personId: 1 }, { personId: 1 }],
+        identifiers: [
+          [{ identifierType: 'EXTERNAL_REL', identifierValue: 'delius1' }],
+          [{ identifierType: 'EXTERNAL_REL', identifierValue: 'delius1' }],
+        ],
+        selected: 'delius1',
       },
     ]
 
     sortingScenarios.forEach(scenario => {
       it(`should select person id where ${scenario.label}`, async () => {
         nomisClientMock.getRoRelations.resolves(scenario.relationships)
-        await service.getResponsibleOfficer('123', 'username')
-        expect(nomisClientMock.getPersonIdentifiers).to.be.calledWith(scenario.selected)
+
+        scenario.identifiers.forEach((response, index) => {
```

Comment:
This is a bit complex can you explain what your testing here?

Reply:
This is stubbing different responses for each call, ie if there are 3 relationships it makes the next call 3 times and I need a different response for each call

Reply:
Would it be clearer to use stub.withArgs(blah).resolves(responseForBlah)?

Reply:
Re-written more explicitly